jfr-agent-extension/build.gradle.kts

```diff
@@ -23,6 +23,7 @@ tasks.shadowJar {
     manifest {
         attributes(
                 "Premain-Class" to "com.newrelic.jfr.Entrypoint",
+                "Main-Class" to "com.newrelic.jfr.StatsMaker",
```

Comment:
My understand is that `jfr-agent-extension` exists only to produce a Java agent extension and that all other use cases (JFR daemon and JFR agent) are handled by the jfr-daemon or jlink artifacts. We're hoping to do away with the entire `jfr-agent-extension` module since we're essentially pulling the `jfr-daemon` artifact into the New Relic Java agent and building a service around it. Does it make sense to add this new functionality here or the daemon artifact?

I guess I don't fully understand the use case. As the PR currently stands, this would produce an agent extension jar  that could also be directly invoked to run the `StatsMaker`?

Am I just fundamentally misunderstanding something?